src/mapping/NearestProjectionMapping.cpp

```diff
@@ -36,10 +36,12 @@ NearestProjectionMapping::NearestProjectionMapping(
     setInputRequirement(Mapping::MeshRequirement::VERTEX);
     setOutputRequirement(Mapping::MeshRequirement::FULL);
   } else {
-    PRECICE_ASSERT(constraint == SCALEDCONSISTENT, constraint);
+    PRECICE_ASSERT(constraint == SCALED_CONSISTENT || constraint == SCALED_CONSISTENT_VOLUME, constraint);
     setInputRequirement(Mapping::MeshRequirement::FULL);
     setOutputRequirement(Mapping::MeshRequirement::FULL);
   }
+
+  PRECICE_CHECK(constraint != SCALED_CONSISTENT_VOLUME, "Nearest-projection can't be used with volume version of the scaled-consistent mapping. Use scaled-consistent instead.");
```

Comment:
Why?

Reply:
It's not technically impossible, but it's semantically wrong. If you use NP, you should only configure surface primitives. Then, scaling over a volume would be weird. If you want a volume-scaling and have connectivity, linear cell interpolation should be used, not NP.

Similarly, surface scaling is not allowed with linear cell interpolation.